cmd/kraft/x/probe/probe.go

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2023, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file expect in compliance with the License.
+package probe
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"kraftkit.sh/cmdfactory"
+	"kraftkit.sh/internal/tableprinter"
+	"kraftkit.sh/iostreams"
+	"kraftkit.sh/log"
+	"kraftkit.sh/packmanager"
+	"kraftkit.sh/unikraft/app"
+
+	"github.com/spf13/cobra"
+)
+
+type Probe struct {
+	Dumpdir string `long:"dir" short:"d" usage:"Choose directory to dump information to (default: workdir)"`
+	Force   bool   `long:"force" short:"f" usage:"Overwrite existing probe.out"`
+	Raw     bool   `long:"raw" short:"r" usage:"Don't try to parse the ELF file, just dump the raw data"`
+}
+
+func New() *cobra.Command {
+	cmd, err := cmdfactory.New(&Probe{}, cobra.Command{
+		Short: "Read Unikraft and library metadata from an ELF binary.",
+		Use:   "probe [FLAGS]",
+		Args:  cobra.ExactArgs(1),
+		Annotations: map[string]string{
+			cmdfactory.AnnotationHelpGroup: "experimental",
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func (opts *Probe) Pre(cmd *cobra.Command, args []string) error {
+	ctx, err := packmanager.WithDefaultUmbrellaManagerInContext(cmd.Context())
+	if err != nil {
+		return err
+	}
+
+	cmd.SetContext(ctx)
+
+	return nil
+}
+
+func (opts *Probe) Run(cmd *cobra.Command, args []string) error {
+	// If a directory is specified, change to that directory
+	// Otherwise, use the current working directory
+	var filepath string
+	if opts.Dumpdir != "" {
+		filepath = path.Join(opts.Dumpdir, "probe.out")
+	} else {
+		filepath = path.Join(".", "probe.out")
+	}
```

Comment:
Is it possible to go-away with output? Is it possible to just serialize the Kraftfile in memory instead?